examples/dark-mode/app/routes/action/set-theme.tsx

```diff
@@ -17,11 +17,22 @@ export const action: ActionFunction = async ({ request }) => {
     });
   }
 
+  if (redirectTo && typeof redirectTo !== "string") {
+    return json({
+      success: false,
+      message: `redirect URL should be a string`,
+    });
+  }
+
   themeSession.setTheme(theme);
-  return json(
-    { success: true },
-    { headers: { "Set-Cookie": await themeSession.commit() } }
-  );
+
+  const headers = { "Set-Cookie": await themeSession.commit() };
+
+  if (redirectTo) {
+    return redirect(redirectTo, { headers });
+  }
+
+  return json({ success: true }, { headers });
 };
 
 export const loader: LoaderFunction = () => redirect("/", { status: 404 });
```

Comment:
```suggestion
export const loader: LoaderFunction = () => redirect("/", { status: 302 });
```

Reply:
Hm, what's the reasoning behind this? This will only trigger when people type this in the URL bar, I thought that it made sense to return 404 considering that there's nothing there to load, only submit, but that's internal and none of their business 😜

Reply:
Not sure either, but a redirect with a 404 code clearly isn't the way 😅

Reply:
That's a good point 😅